### PR TITLE
PREAPPS-1180: "Run" action for existing filters 

### PR DESCRIPTION
**Pull request by craigdanj**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2036,7 +2036,7 @@
     },
```
Comment (madhavdhuppe) [inline package-lock.json]: I think we are not updating package-lock.json files if we haven’t updated/added npm packages.

Comment (craigdanj) [reply package-lock.json]: you are right. This was done by mistake. Will undo it.

Comment (craigdanj) [reply package-lock.json]: Done

File: src/batch-client/index.ts [hunk 2]
```diff
@@ -247,6 +248,28 @@ export class ZimbraBatchClient {
...
+			body: {
+				filterRules: {
+					filterRule: filterRules
+				},
+				m: {
+					ids
+				}
+			}
+		}).then(res => {
+			let ids = get(res, 'm[0].ids');
```
Comment (silentsakky) [inline src/batch-client/index.ts]:
```
const ids = get(res, 'm[0].ids');

			if (ids) {
				return ids.split(',');
			}

			return [];
```

Comment (craigdanj) [reply src/batch-client/index.ts]: done

File: src/batch-client/index.ts [hunk 3]
```diff
@@ -247,6 +248,27 @@ export class ZimbraBatchClient {
...
+					filterRule: filterRules
+				},
+				m: {
+					ids
+				}
+			}
+		}).then(res => {
+			const ids = get(res, 'm[0].ids');
+
+			if (ids) {
```
Comment (avadhutprabhudesai) [inline src/batch-client/index.ts]: return ids ? ids.split(',') : [];

Comment (craigdanj) [reply src/batch-client/index.ts]: done

File: src/batch-client/index.ts [hunk 4]
```diff
@@ -241,6 +242,22 @@ export class ZimbraBatchClient {
...
+			name: 'ApplyFilterRules',
+			body: {
+				filterRules: {
+					filterRule: filterRules
+				},
+				m: {
+					ids
+				}
+			}
+		}).then(res => {
```
Comment (itsmeprashant) [inline src/batch-client/index.ts]: then(res => get(res, 'm[0].ids', []).split(','));

Comment (itsmeprashant) [reply src/batch-client/index.ts]: never mind it will not work, I thought of list when I commented.

**Pull request merged**